src/tools/gxpkg.ss

```diff
@@ -523,13 +592,91 @@
             (build.ss (pkg-build-script pkg)))
        (displayln "... build " pkg)
+       (pkg-manifest! pkg)
        (run-process [build.ss "compile" build-options ...]
                     directory: path
                     coprocess: void
                     stdout-redirection: #f)
        (when dependents?
          (for-each pkg-build (pkg-dependents pkg)))))))
 
+(def (pkg-manifest! pkg)
+  (let* (((values pkg _) (pkg+tag pkg))
+         (plist (pkg-plist pkg))
+         (deps (pgetq depend: plist []))
+         (deps
+          (let recur ((rest deps) (result []))
+            (match rest
+              ([dep . rest]
+               (let ((values dep _) (pkg+tag dep))
+                 ;; check for external package manager installed deps (eg NiX)
+                 (if (file-exists? (pkg-plist-path dep))
+                   (let* ((plist (pkg-plist dep))
+                          (deps (pgetq depend: plist [])))
+                     (recur rest (recur deps (cons dep result))))
+                   ;; just record the dep, we don't have the pkg contents
+                   ;; for transitive
+                   (recur rest (cons dep result)))))
+              (else
+               (remove-duplicates result)))))
+         (manifests
+          (let lp ((rest deps) (result []))
+            (match rest
+              ([dep . rest]
+               (let (manifest
+                     (call-with-input-file
+                         (path-expand (string-append dep ".manifest")
+                           (pkg-root-dir))
+                       read))
+                 (lp rest (append result manifest))))
+              (else
+               (remove-duplicates result)))))
+         (gerbil-version
+          (cons "Gerbil"(gerbil-version-string)))
+         (gambit-version
+          (cons "Gambit" (system-version-string)))
+         (write-version-manifest
+          (lambda (manifest1 output)
+            (pretty-print
+             `(def version-manifest
+                (quote
+                 ,(remove-duplicates
+                   (cons* manifest1 gerbil-version gambit-version manifests))))
+             output)))
+         (write-pkg-manifest
+          (lambda (manifest1 output)
+            (pretty-print
+             (remove-duplicates (cons* manifest1 gerbil-version gambit-version manifests))
+             output))))
+
+    (if (equal? pkg ".")
+      (let* ((version
+              (if (file-exists? ".git")
+                (run-process ["git" "describe" "--tags" "--always"]
+                             coprocess: read-line)
+                "unknown"))
+             (manifest1
+              (cons (path-strip-directory
+                     (let (path (path-normalize (current-directory)))
+                       (substring path 0 (1- (string-length path)))))
+                    version)))
+        (call-with-output-file [path: "manifest.ss" create: 'maybe truncate: #t]
```

Comment:
In what module path prefix? Won't there be clashes?

Reply:
at the top of the current package for inclusion, not import.

So no namespace issues.